modules/samples/watertank/component/api/src/main/java/org/homeautomation/watertank/api/DeviceTypeServiceImpl.java

```diff
@@ -0,0 +1,434 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+
+/**
+ * This is the API which is used to control and manage device type functionality
+ */
+@SuppressWarnings("NonJaxWsWebServices")
+@API(name = "watertank", version = "1.0.0", context = "/watertank", tags = "watertank")
+@DeviceType(value = "watertank")
+public class DeviceTypeServiceImpl implements DeviceTypeService {
+
+    private static final String KEY_TYPE = "PRODUCTION";
+    private static Log log = LogFactory.getLog(DeviceTypeService.class);
+    private static ApiApplicationKey apiApplicationKey;
+    private ConcurrentHashMap<String, DeviceJSON> deviceToIpMap = new ConcurrentHashMap<>();
+
+    private static String shortUUID() {
+        UUID uuid = UUID.randomUUID();
+        long l = ByteBuffer.wrap(uuid.toString().getBytes(StandardCharsets.UTF_8)).getLong();
+        return Long.toString(l, Character.MAX_RADIX);
+    }
+
+    /**
+     * @param agentInfo device owner,id
+     * @return true if device instance is added to map
+     */
+    @Path("device/register")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response registerDevice(final DeviceJSON agentInfo) {
+        String deviceId = agentInfo.deviceId;
+        if ((agentInfo.deviceId != null) && (agentInfo.owner != null)) {
+            deviceToIpMap.put(deviceId, agentInfo);
+            return Response.status(Response.Status.OK).build();
+        }
+        return Response.status(Response.Status.NOT_ACCEPTABLE).build();
+    }
+
+    /**
+     /**
+     * @param deviceId unique identifier for given device type
+     * @param onLevel  level to turn on the relay
+     * @param offLevel level to turn off thr relay
+     */
+    @Path("device/{deviceId}/change-levels")
+    @POST
+    @Feature(code = "change-levels", name = "Change on/off water levels",
+            description = "Change on/off water levels")
+    public Response changeOnOffLevels(@PathParam("deviceId") String deviceId,
+                                      @QueryParam("on") int onLevel,
+                                      @QueryParam("off") int offLevel,
+                                      @Context HttpServletResponse response) {
+        try {
+            if (!APIUtil.getDeviceAccessAuthorizationService()
+                    .isUserAuthorized(new DeviceIdentifier(deviceId, DeviceTypeConstants.DEVICE_TYPE))) {
+                return Response.status(Response.Status.UNAUTHORIZED.getStatusCode()).build();
+            }
+
+            JSONObject jsonObject = new JSONObject();
+            jsonObject.put("on", onLevel);
+            jsonObject.put("off", offLevel);
+
+            Map<String, String> dynamicProperties = new HashMap<>();
+            String publishTopic = APIUtil.getAuthenticatedUserTenantDomain()
+                                  + "/" + DeviceTypeConstants.DEVICE_TYPE + "/" + deviceId + "/command";
```

Comment:
When a line is broken at a non-assignment operator the break comes before the symbol. 
https://google.github.io/styleguide/javaguide.html#s4.5.1-line-wrapping-where-to-break